### DISCUSSION TITLE
How do you test middleware?

### DISCUSSION THREAD

**Discussion by brandonk3nt** [RobinTail/express-zod-api#1908] [Q&A] [answered]
I have a standard middleware implementation like this:
```
const authMiddleware = new Middleware({ ...
```
How can I write tests for the handler?

**Comment by RobinTail** [comment]
in general, this way, @brandonk3nt : 

```ts
expect(await authMiddleware.execute({
  input: {},
  options: {},
  request: requestMock,
  response: responseMock,
  logger: loggerMock
}).toEqual({ })
```

however, you'd have to make those mocks.
There is no helper for testing middlewares yet — only `testEndpoint` (with all its middlewares).
Perhaps I should make one.

**Reply by brandonk3nt** [reply]
awesome, thanks!

**Reply by RobinTail** [reply]
forgot `await`, or can use `.resolves.`, depends on testing framework

**Comment by RobinTail** [comment]
I started making the helper for that purpose in https://github.com/RobinTail/express-zod-api/pull/1909

**Comment by RobinTail** [comment, marked as answer]
The feature is available in the recently released v20.4.0, @brandonk3nt
